From 1c5ec5c436ca46f40353410da0b24dff5c0145ee Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Thu, 20 Jun 2019 14:57:52 +0200
Subject: [PATCH] bcmdhd: Fix wifi disconnect problems

Patch ported from https://devtalk.nvidia.com/default/topic/
1047138/jetson-tx1/wifi-disconnect-problem-on-jetpack-3-3/2

Upstream-status: Backport
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 drivers/net/wireless/bcmdhd/wl_cfg80211.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
index 9d3568d..8f5f11d 100644
--- a/drivers/net/wireless/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
@@ -9935,6 +9935,7 @@ wl_cfg80211_verify_bss(struct bcm_cfg80211 *cfg, struct net_device *ndev)
 	do {
 		bss = CFG80211_GET_BSS(wiphy, NULL, curbssid,
 			ssid->SSID, ssid->SSID_len);
+		cfg->wdev->ssid_len = ssid->SSID_len;
 		if (bss || (count > 5)) {
 			break;
 		}
-- 
2.7.4

